PullRequest for Issue126: Refactor the dwell implemention for Amptek

XuetaoLiu-S opened the pull request:
https://github.com/acquaman/AcquamanDataServer/issues/126

davidChevrier committed:
First working implementation of using internal flattening request for Amptek dwell. Appears to be working and writing data to the spectrum PV. Now a lot of testing and code cleanup is needed to make it easier to follow the logic and remove unnecessary features.

davidChevrier committed:
First set of changes to comment out unnecessary code blocks. Tested with both trigger acquistions and continuous scans, appears to be working.

davidChevrier committed:
Second set of changes to comment out unnecessary code. Testing has been succesful. Need to update a handful of items for the Amptek dwell trigger.

davidChevrier committed:
Missed a pair of member variables that could be removed from AmptekSDD123EPICSDetectorManager.

XuetaoLiu-S committed:
small code refactor to shift codes around and release the timer

XuetaoLiu-S committed:
delete the internal request when done and use currentDateTimeUTC() instead of currentDateTime()

XuetaoLiu-S committed:
Remove unused codes

XuetaoLiu-S commented:
@davidChevrier 
I removed the commented codes and make some small changes. Pls take a look.

XuetaoLiu-S commented:
Tested OK

dretrex commented:
Looks okay to me.  @davidChevrier should probably still go over it as well.
Time Estimate: 5 minutes

XuetaoLiu-S committed:
Update AMDSCentralServerSGMAmptek.cpp

The request object to be deleted should be `clientRequest` instead of `returnedRequest`

davidChevrier commented on source/Detector/Amptek/SGM/AmptekSDD123EPICSDetectorManager.h:
```diff
@@ -63,11 +63,11 @@ protected slots:
 protected:
 	/// helper function to handle continuous spectrum data update
-	void dataHelper(AMDSDataHolder *spectrum, int count, double elapsedTime);
+	void dataHelper(AMDSDataHolder *spectrum, double count, double elapsedTime);
```
Why did the count parameter change to double here?

XuetaoLiu-S replied:
There is a compile warning. in the following codes, we passed in a double.

```
void AmptekSDD123EPICSDetectorManager::setFlattenedSpectrumData(AMDSDataHolder *dataHolder)
  {
    connect(spectrumControl_, SIGNAL(valueChanged(double)), this, SLOT(onSpectrumControlValueChanged()));

    double elapsedTime = dwellTime();
    double count = elapsedTime*1000/50;
    dataHelper(dataHolder, count, elapsedTime);
  }
```

dretrex replied:
Isn't count the number of data positions?  That should definitely be an int if that's the case.

davidChevrier commented:
Overall, it looks very good to me. Thanks for catching a bunch of issues @XuetaoLiu-CLS 
I think this is ready for testing, but we'll have to wait to make sure that SGM comes up in a stable condition before actually doing these tests.

Time Estimate: 20 minutes

XuetaoLiu-S commented:
@davidChevrier I tested it yesterday, the step scan works as expected.

XuetaoLiu-S committed:
change the dataHelper signature back and convert count to int

XuetaoLiu-S committed:
Merge branch 'Issue126' of https://github.com/acquaman/AcquamanDataServer into Issue126

davidChevrier commented:
@iainworkman 
I'm about to merge this into master. Is there anything we need to be cognizant of when we want get this in as a release candidate?

iainworkman commented:
No, I will bear it in mind when I'm doing the deploy. It'll affect the test procedures which I perform (ensuring the dwells work as expected, and the step scanning is okay).

XuetaoLiu-S commented:
@iainworkman we can deploy after this and #129 together so that we can deploy the AMDSPV at the same time

davidChevrier commented:
@iainworkman @XuetaoLiu-CLS 
I think it would be fairly reasonable to wait to deploy this with #129 
I think that SGM needs some time and user shifts to demonstrate they are happy with the current deployment before we go again. @iainworkman could you check into the user schedule and see what the next 2-3 weeks look like?
